scripts/tests/run_test_suite.py

```diff
@@ -396,102 +399,118 @@ def build_app(arg_value, kind: str, key: str):
 
     ble_controller_app = None
     ble_controller_tool = None
+    to_terminate: list[Terminatable] = []
 
-    if sys.platform == 'linux':
-        ns = chiptest.linux.IsolatedNetworkNamespace(
-            index=0,
-            # Do not bring up the app interface link automatically when doing BLE-WiFi commissioning.
-            setup_app_link_up=not ble_wifi,
-            # Change the app link name so the interface will be recognized as WiFi or Ethernet
-            # depending on the commissioning method used.
-            app_link_name='wlx-app' if ble_wifi else 'eth-app',
-            unshared=context.obj.in_unshare)
-
-        if ble_wifi:
-            bus = chiptest.linux.DBusTestSystemBus()
-            bluetooth = chiptest.linux.BluetoothMock()
-            wifi = chiptest.linux.WpaSupplicantMock("MatterAP", "MatterAPPassword", ns)
-            ble_controller_app = 0   # Bind app to the first BLE controller
-            ble_controller_tool = 1  # Bind tool to the second BLE controller
-
-        executor = chiptest.linux.LinuxNamespacedExecutor(ns)
-    elif sys.platform == 'darwin':
-        executor = chiptest.darwin.DarwinExecutor()
-    else:
-        log.warning("No platform-specific executor for '%s'", sys.platform)
-        executor = Executor()
-
-    runner = chiptest.runner.Runner(executor=executor)
-
-    log.info("Each test will be executed %d times", iterations)
-
-    apps_register = AppsRegister()
-    apps_register.init()
+    def cleanup() -> None:
+        for item in reversed(to_terminate):
+            try:
+                log.info("Cleaning up %s", item.__class__.__name__)
+                item.terminate()
+            except Exception as e:
+                log.warning("Encountered exception during cleanup: %s", e)
```

Comment:
It's best to print exceptions as %r as then the exception type is also printed. Also for exception handlers it's good to add exc_info=True to print the stack trace.

Reply:
Fixed